File: ProjecIntegration.Api/ProjecIntegration.Api/Models/Entity/Complexe.cs
using ProjecIntegration.Api.Models;

namespace ProjecIntegration
{
    public class Complexe :BaseEntity
    {
       
        public string Name { get; set; }
        public string Description { get; set; }
        public string Adress { get; set; }
        public List<SalleDeTheatre> SalleDeTheatres { get; set; }
        //public Entreprise Entreprise { get; set; }
        //public int EntrepriseID { get; set; }
    }
}

File: projectintegrationFront/src/app/class/complexe.ts
import { SalleDeTheatre } from "./salledetheatre";

export class Complexe {
  id: number;
  name: string;
  description: string;
  adress: string;
  salleDeTheatres: SalleDeTheatre[];


  constructor(
    id: number,
    name: string,
    description: string,
    adress: string,
    salleDeTheatres: SalleDeTheatre[]

  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.adress=adress
    this.salleDeTheatres = salleDeTheatres;
  }
}
export class AddComplexe{
    name: string;
    description: string;
  address: string;
  constructor(
        name: string,
        description: string,
        address: string
  ) {
    this.name = name;
    this.description = description;
    this.address=address
    }
}

// UpdateComplexeDto
export class UpdateComplexe{
    id: number;
    name: string;
    description: string;
    address: string;
  constructor(
    id:number,
    name: string,
    description: string,
    address: string
  )
  { 
    this.id = id;
    this.name = name;
    this.description = description;
    this.address = address;
  }
}
